openthread_border_router/config.yaml

```diff
@@ -37,7 +37,7 @@ ports_description:
   8080/tcp: OpenThread Web port
   8081/tcp: OpenThread REST API port
 schema:
-  device: device(subsystem=tty)?
+  device: device(subsystem=tty)
```

Comment:
Hm, but does this work now? I mean this will mean you need to specify a device in options now? Feels a bit weird.

Maybe optional and drop the null in default? :thinking:

Reply:
Actually, we use this combination in other places, e.g. Multiprotocol: https://github.com/home-assistant/addons/blob/master/silabs-multiprotocol/config.yaml#L27 :thinking:

Reply:
Hm, I see. From your git commit message ("this doesnt work in conjunction with null default.") I was under the impression it rejected the config or something, which made me wonder.

But yeah I see, even though it is optional, the UI (even when using the YAML editor) rejects null and requires a device to be specified. So leaving it non-optional makes this a bit more explicit (and also doesn't add a unnecessary change. :+1:

Reply:
> You need to select a random tty/com port, and the network settings override that.

Thank you for the hint. That's a really strange behaviour.

Just tested the network_mode with the SLZB-06 (2.1.3.dev). I receive `Failed to communicate with RCP - no response from RCP during initialization`. What is the default port? 6638 is displayed in your UI of the device.

Reply:
> Just tested the network_mode with the SLZB-06 (2.1.3.dev). I receive `Failed to communicate with RCP - no response from RCP during initialization`. What is the default port? 6638 is displayed in your UI of the device.

default port is 6638 or whatever is set in web ui, however:
- make sure you set baudrate to `460800` on the zha/z2m page in web ui
- I not sure it works with `.local` addresses try instead `ip:port`

If you still have issues file a new issue with logs.

Reply:
@tl-sl Thanks, looks like there was an issue in core firmware 2.1.3.dev - 2.1.5.dev looks more promising:

```
-----------------------------------------------------------
 Add-on: OpenThread Border Router
 OpenThread Border Router add-on
-----------------------------------------------------------
 Add-on version: 2.6.0
 You are running the latest version of this add-on.
s6-rc: info: service socat-otbr-tcp successfully started
 System: Home Assistant OS 12.2  (amd64 / generic-x86-64)
 Home Assistant Core: 2024.5.1
 Home Assistant Supervisor: 2024.04.4
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service banner successfully started
s6-rc: info: service universal-silabs-flasher: starting
[15:14:37] INFO: Flashing firmware is disabled
s6-rc: info: service universal-silabs-flasher successfully started
s6-rc: info: service otbr-agent: starting
[15:14:38] INFO: Setup OTBR firewall...
[15:14:38] INFO: Starting otbr-agent...
otbr-agent[189]: [NOTE]-AGENT---: Running 0.3.0-2279c02-dirty
otbr-agent[189]: [NOTE]-AGENT---: Thread version: 1.3.0
otbr-agent[189]: [NOTE]-AGENT---: Thread interface: wpan0
otbr-agent[189]: [NOTE]-AGENT---: Radio URL: spinel+hdlc+uart:///tmp/ttyOTBR?uart-baudrate=460800
otbr-agent[189]: [NOTE]-AGENT---: Radio URL: trel://eno1
otbr-agent[189]: [NOTE]-ILS-----: Infra link selected: eno1
otbr-agent[189]: 00:00:00.044 [N] RoutingManager: No valid /48 BR ULA prefix found in settings, generating new one
otbr-agent[189]: 00:00:00.050 [N] RoutingManager: BR ULA prefix: fd67:8176:9626::/48 (generated)
otbr-agent[189]: 00:00:00.050 [N] RoutingManager: Local on-link prefix: fdde:ad00:beef:cafe::/64
otbr-agent[189]: 00:00:00.078 [W] Platform------: [netif] Failed to process request#2: No such process
s6-rc: info: service otbr-agent successfully started
s6-rc: info: service otbr-agent-rest-discovery: starting
s6-rc: info: service otbr-agent-configure: starting
Done
s6-rc: info: service otbr-agent-configure successfully started
otbr-agent[189]: 00:00:00.542 [N] RoutingManager: Local on-link prefix: fd6a:7187:7a06:dca3::/64
otbr-agent[189]: 00:00:00.554 [N] Mle-----------: Role disabled -> detached
otbr-agent[189]: 00:00:00.566 [N] Platform------: [netif] Changing interface state to up.
otbr-agent[189]: 00:00:00.572 [W] Platform------: [netif] ADD [U] fe80:0:0:0:d00a:b39d:d89b:756b failed (InvalidArgs)
otbr-agent[189]: 00:00:00.572 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
otbr-agent[189]: 00:00:00.573 [W] Platform------: [netif] ADD [U] fd48:4894:182c:7c7b:e14e:82a2:611d:28d9 failed (InvalidArgs)
otbr-agent[189]: 00:00:00.573 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
otbr-agent[189]: 00:00:00.573 [W] Platform------: [netif] Failed to process request#5: No such process
[15:14:38] INFO: Successfully sent discovery information to Home Assistant.
s6-rc: info: service otbr-agent-rest-discovery successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
otbr-agent[189]: 00:00:00.709 [N] Mle-----------: Attach attempt 1, AnyPartition reattaching with Active Dataset
otbr-agent[189]: 00:00:07.209 [N] RouterTable---: Allocate router id 3
otbr-agent[189]: 00:00:07.209 [N] Mle-----------: RLOC16 fffe -> 0c00
otbr-agent[189]: 00:00:07.215 [N] Mle-----------: Role detached -> leader
otbr-agent[189]: 00:00:07.215 [N] Mle-----------: Partition ID 0x29e54005
otbr-agent[189]: 00:00:07.233 [W] Platform------: [netif] ADD [U] fd48:4894:182c:7c7b:0:ff:fe00:c00 failed (InvalidArgs)
otbr-agent[189]: 00:00:07.233 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
otbr-agent[189]: 00:00:07.233 [W] Platform------: [netif] ADD [U] fd48:4894:182c:7c7b:0:ff:fe00:fc00 failed (InvalidArgs)
otbr-agent[189]: 00:00:07.233 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
otbr-agent[189]: [NOTE]-BBA-----: BackboneAgent: Backbone Router becomes Primary!
otbr-agent[189]: 00:00:07.537 [W] Platform------: [netif] ADD [U] fd48:4894:182c:7c7b:0:ff:fe00:fc38 failed (InvalidArgs)
otbr-agent[189]: 00:00:07.537 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
otbr-agent[189]: 00:00:07.537 [W] Platform------: [netif] ADD [U] fd48:4894:182c:7c7b:0:ff:fe00:fc10 failed (InvalidArgs)
otbr-agent[189]: 00:00:07.537 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
otbr-agent[189]: 00:00:07.951 [W] Platform------: [netif] ADD [U] fd67:8176:9626:1:eb35:8303:413a:451b failed (InvalidArgs)
otbr-agent[189]: 00:00:07.951 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
otbr-agent[189]: 00:00:17.474 [W] Platform------: [netif] ADD [U] fd48:4894:182c:7c7b:0:ff:fe00:fc11 failed (InvalidArgs)
otbr-agent[189]: 00:00:17.475 [W] Platform------: [netif] Failed to process event, error:InvalidArgs
```